neqo-http3/src/headers_checks.rs

```diff
@@ -30,27 +34,31 @@ pub fn is_interim(headers: &[Header]) -> Res<bool> {
     }
 }
 
-fn track_pseudo(name: &str, state: &mut u8, message_type: MessageType) -> Res<bool> {
+fn track_pseudo(
+    name: &str,
+    state: &mut EnumSet<PseudoHeaderState>,
+    message_type: MessageType,
+) -> Res<bool> {
     let (pseudo, bit) = if name.starts_with(':') {
-        if *state & REGULAR_HEADER != 0 {
+        if state.contains(PseudoHeaderState::None) {
             return Err(Error::InvalidHeader);
         }
         let bit = match (message_type, name) {
-            (MessageType::Response, ":status") => PSEUDO_HEADER_STATUS,
-            (MessageType::Request, ":method") => PSEUDO_HEADER_METHOD,
-            (MessageType::Request, ":scheme") => PSEUDO_HEADER_SCHEME,
-            (MessageType::Request, ":authority") => PSEUDO_HEADER_AUTHORITY,
-            (MessageType::Request, ":path") => PSEUDO_HEADER_PATH,
-            (MessageType::Request, ":protocol") => PSEUDO_HEADER_PROTOCOL,
+            (MessageType::Response, ":status") => PseudoHeaderState::Status,
+            (MessageType::Request, ":method") => PseudoHeaderState::Method,
+            (MessageType::Request, ":scheme") => PseudoHeaderState::Scheme,
+            (MessageType::Request, ":authority") => PseudoHeaderState::Authority,
+            (MessageType::Request, ":path") => PseudoHeaderState::Path,
+            (MessageType::Request, ":protocol") => PseudoHeaderState::Protocol,
```

Comment:
Maybe implement `TryFrom<(MessageType, &str)>` for the enum.

Reply:
Nice suggestion. Thanks.